bundleserver/src/main/java/net/discdd/server/applicationdatamanager/BundleServerAduDeliverer.java

```diff
@@ -59,11 +65,23 @@ synchronized private void revalidateApps() {
             foundApps.add(appAdapter.getAppId());
             // TODO: we should also check if the location matches
             if (!apps.containsKey(appAdapter.getAppId())) {
-                var stub = ServiceAdapterServiceGrpc.newBlockingStub(
-                        ManagedChannelBuilder.forTarget(appAdapter.getAddress()).usePlaintext().build());
-                apps.put(appAdapter.getAppId(),
-                         new AppState(appAdapter.getAppId(), Executors.newSingleThreadExecutor(), new HashSet<>(),
-                                      stub));
+                try {
+                    var sslClientContext = GrpcSslContexts.forClient()
+                            .keyManager(serverSecurity.getJavaKeyPair().getPrivate(), serverSecurity.getServerCert())
+                            .trustManager(DDDTLSUtil.trustManager)
+                            .build();
+                    var channel = NettyChannelBuilder.forTarget(appAdapter.getAddress())
+                            .useTransportSecurity()
+                            .sslContext(sslClientContext)
+                            .build();
+
+                    var stub = ServiceAdapterServiceGrpc.newBlockingStub(channel);
+                    apps.put(appAdapter.getAppId(),
+                            new AppState(appAdapter.getAppId(), Executors.newSingleThreadExecutor(), new HashSet<>(),
+                                    stub));
+                } catch (SSLException e) {
+                    throw new RuntimeException(e);
```

Comment:
put a log in and skip this adapter. if you throw an exception, you can mess up other adapters.